src/endpointsApi.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const logger = require('./logger')
+const EndpointModel = require('./models/endpoints')
+const {handleServerError, validateEndpoint} = require('./util')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const saveEndPoint = body => {
+  return new Promise((resolve, reject) => {
+    const endpoint = new EndpointModel(body)
+    endpoint.save(err => {
+      if (err) reject(err)
+      resolve(endpoint)
+    })
+  })
+}
+
+const updateEndPoint = (endpointId, body) => {
+  return new Promise((resolve, reject) => {
+    EndpointModel.findOneAndUpdate(
+      {_id: endpointId},
+      body,
+      {new: true, runValidators: true},
+      (err, result) => {
+        if (err) reject(err)
+        resolve(result)
+      }
+    )
+  })
+}
+
+const createCreateEndpoint = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
+
+    try {
+      await KoaBodyParser()(ctx, () => {})
+
+      if (
+        !ctx.request ||
+        !ctx.request.body ||
+        !Object.keys(ctx.request.body).length
+      ) {
+        ctx.status = 400
+        const err = `${failureMsg}Invalid endpoint object`
+        ctx.body = {error: err}
+        logger.error(err)
+        return next()
+      }
+      const body = Object.assign({}, ctx.request.body, {
+        lastUpdated: Date.now()
+      })
+      const inValid = validateEndpoint(body)
```

Comment:
@bradsawadye / @BMartinos Do the names or paths of these endpoints need to be unique?

Things could very easily become confusing if you send through an endpoint creation twice and then try updating your endpoint config.

Reply:
That is a good point, and yes they would need to be unique. So when trying to create a new endpoint we will need to verify that pattern doesnt already exist otherwise incorrect mapping could be triggered

Reply:
the patterns have to be unique, this should be set inthe mongoose schema

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/48/commits/9e6a514a66f3fe013386a6243286c5e78f86b761